common/containers/Tabs/SendTransaction/index.tsx

```diff
@@ -292,7 +288,20 @@ export class SendTransaction extends React.Component<Props, State> {
   }
 
   public parseQuery() {
-    const query = this.props.location.query;
+    const search = this.props.location.search;
+    const query = search
+      ? JSON.parse(
+          '{"' +
+            search
+              .substring(1)
+              .replace(/&/g, '","')
+              .replace(/=/g, '":"') +
+            '"}',
+          (key, val) => {
+            return key === '' ? val : decodeURIComponent(val);
+          }
+        )
+      : {};
```

Comment:
I get what this is doing, but it feels like black magic that's prone to throwing depending on what's in the search query, and doesn't allow for typing. What happened to URLSearchParams?

Reply:
Your right, it's a bit hackey. Since there will likely be other situations where parsing a query string in a url is useful, I'm writing a few util functions to fix that. I'll change this to WIP till i get that sorted.